documentation/library-reference/source/coloring-stream/index.rst

```diff
@@ -0,0 +1,221 @@
+   :value well?: An instance of :drm:`<boolean>`.
+
+   :description:
+
+     Return whether or not the underlying stream supports color
+     output.
+
+     This checks that:
+
+     * The stream is a :class:`<file-stream>`.
+     * :gf:`stream-console?` is true. (On Unix, this means that ``isatty``
+       is true for the stream.)
+     * The ``TERM`` environment variable is not ``"dumb"``.
+     * The ``EMACS`` environment variable is not ``"t"``.
+
+Text Attributes
+===============
+
+.. class:: <text-attributes>
+
+   :superclasses: :drm:`<object>`
+
+   :keyword background: An instance of ``false-or(<text-color>)``.
+   :keyword foreground: An instance of ``false-or(<text-color>)``.
+   :keyword intensity: An instance of ``false-or(<text-intensity>)``.
+
+   :description:
+
+     Instances of ``<text-attributes>`` are used for representing the
+     desired text appearance. They can be passed to :gf:`format` when
+     writing to a :class:`<coloring-stream>`.
+
+     *background* and *foreground*, if given, should be one of the color
+     constants like :const:`$color-red`, :const:`$color-green`, etc.
+
+     *intensity*, if given, should be one of :const:`$bright-text-intensity`,
+     :const:`$dim-color-intensity` or :const:`$normal-text-intensity`.
+
+     Values that are omitted are set to the default values for the terminal.
+
+.. function:: text-attributes
+
+   :signature: text-attributes (#key foreground background intensity) => (attributes)
+
+   :parameter #key foreground: An instance of ``false-or(<text-color>)``.
+   :parameter #key background: An instance of ``false-or(<text-color>)``.
+   :parameter #key intensity: An instance of ``false-or(<text-intensity>)``.
+   :value attributes: An instance of :class:`<text-attributes>`.
+
+   :description:
+
+     ``text-attributes`` provides an easy wrapper for creating instances
+     of :class:`<text-attributes>`.
+
+     *background* and *foreground*, if given, should be one of the color
+     constants like :const:`$color-red`, :const:`$color-green`, etc.
+
+     *intensity*, if given, should be one of :const:`$bright-text-intensity`,
+     :const:`$dim-color-intensity` or :const:`$normal-text-intensity`.
+
+     Values that are omitted are set to the default values for the terminal.
+
+   :example:
+
+     .. code-block:: dylan
+
+        let error-text = text-attributes(foreground: $color-red);
+
+.. constant:: $text-reset
+
+   :type: :class:`<text-attributes>`
+
+   :description:
+
+     A predefined constant to reset the text back to the default attributes.
+
+     This is equivalent to ``text-attributes(intensity: $normal-text-intensity)``.
+
+Text Intensity
+--------------
+
+.. constant:: $bright-text-intensity
+
+.. constant:: $dim-text-intensity
+
+   :description:
+
+     .. note:: Not all terminals support dimmed text.
+
+.. constant:: $normal-text-intensity
+
+Text Colors
+-----------
+
+.. constant:: $color-black
```

Comment:
I've been thinking about this as well. One argument that is either for or against, I'm not sure which, is that DUIM and Deuce have constants named ``$red`` and so on.